rewriter/Minitest.cc

```diff
@@ -215,18 +227,58 @@ ast::ExpressionPtr runUnderEach(core::MutableContext ctx, core::NameRef eachName
     return stmt;
 }
 
+bool isDestructuringArg(core::GlobalState &gs, const ast::MethodDef::ARGS_store &args, const ast::ExpressionPtr &expr) {
+    auto *local = ast::cast_tree<ast::UnresolvedIdent>(expr);
+    if (local == nullptr || local->kind != ast::UnresolvedIdent::Kind::Local) {
+        return false;
+    }
+
+    auto name = local->name;
+    if (name.kind() != core::NameKind::UNIQUE || name.dataUnique(gs)->original != core::Names::destructureArg()) {
+        return false;
+    }
+
+    return absl::c_find_if(args, [name](auto &argExpr) {
+               auto *arg = ast::cast_tree<ast::UnresolvedIdent>(argExpr);
+               return arg && arg->name == name;
+           }) != args.end();
+}
+
+// Destructuring blocks are added to method/block entries as an InsSeq where the method body is the distinguished final
+// statement, and the statements of the InsSeq are all individual InsSeq expressions that make up the destructuring for
+// a specific argument.
```

Comment:
```suggestion
// When given code that looks like
//
//     test_each(pairs) do |(x, y)|
//       # ...
//     end
//
// Sorbet desugars it to essentially
//
//     test_each(pairs) do |<destructureArg$1|
//       x = <destructureArg$1>[0]
//       y = <destructureArg$1>[1]
//
//       # ...
//     end
//
// which would otherwise defeat the "Only valid it-blocks can appear within test_each" error message.
//
// Because this case is so common, we have special handling to detect "contains only valid it-blocks
// plus desugared destruturing assignments."
```

it took me longer than I would have liked to figure out why this change works. If this had been the comment here, I think it would have made sense much faster. (Also, maybe I'm just rusty)